Add auth & app password docs and clean up README (#50)
bluesky-social/atproto-website: content/specs/atp.md

### Authentication

Authentication is a simple session-oriented process. View the API call in the applications model section of the docs [here](https://atproto.com/guides/applications#signing-in).

#### App passwords

> We also have app passwords, a short-term solution for authentication that will let users experiment with third-party clients without having to fully trust them with their passwords. In the long term, we plan to implement SSO (Single Sign-On) with scoped permissions.

Reviewer: ```suggestion
We also have app passwords, an initial solution for authentication that will let users use third-party clients without having to trust them with their primary password. In the long term, we plan add SSO (Single Sign-On) authentication with scoped permissions.
```

Users can log into third-party apps with an app password in the same way that they login with their account password. App passwords have most of the same abilities as the user's account password, but they're restricted from destructive actions such as account deletion or account migration. They are also restricted from creating additional app passwords.

> No client changes are required to adopt app passwords. However, we strongly encourage you to prompt users to use an app password on login and avoid ever entering their password. For account creation, we encourage redirecting a user to the bluesky client (https://staging.bsky.app).

Reviewer: ```suggestion
No client changes are required to adopt app passwords. However, we strongly encourage you to prompt users to use an app password on login and avoid ever entering their password. For account creation, we encourage redirecting a user to the [Bluesky web client](https://staging.bsky.app).
```

> We also strongly encourage you to delete all existing refresh tokens and re-fetch access/refresh tokens using an app password.

Reviewer 2: It took me a second to parse this.  I think was I was missing was just the context that this is for clients who are migrating from accepting primary passwords to app passwords.  Maybe a small tweak like,
```suggestion
If you expect users have used their primary password with your client, we also strongly encourage you to delete all existing refresh tokens and re-fetch access/refresh tokens using an app password.
```
